entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EnrichmentCountQueryService.java

```diff
@@ -0,0 +1,82 @@
+package eu.europeana.entitymanagement.web.service;
+
+
+import eu.europeana.entitymanagement.common.config.EntityManagementConfiguration;
+import eu.europeana.entitymanagement.exception.ScoringComputationException;
+import eu.europeana.entitymanagement.web.model.EnrichmentCountResponse;
+import java.time.Duration;
+import java.time.Instant;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Service
+public class EnrichmentCountQueryService {
+
+  private final WebClient webClient;
+  private final String searchApiUriPrefix;
+  private final String enrichmentQuery;
+  private final String enrichmentQueryOrgs;
+
+  private static final Logger logger = LogManager.getLogger(EnrichmentCountQueryService.class);
+  private static final String ERROR_MSG = "Error retrieving enrichmentCount for entityId=";
+
+
+  public EnrichmentCountQueryService(EntityManagementConfiguration configuration) {
+    searchApiUriPrefix = configuration.getSearchApiUrlPrefix();
+    enrichmentQuery = configuration.getEnrichmentsQuery();
+    enrichmentQueryOrgs = configuration.getEnrichmentsQueryOrgs();
+
+    webClient = WebClient.builder().build();
+  }
+
+
+  /**
+   * Queries the Search API to retrieve enrichment counts for the given entityId. Organizations have
+   * a different query string.
+   *
+   * @param entityId       entityID string
+   * @param isOrganization indicates whether this is an organization.
+   */
+  public int getEnrichmentCount(String entityId, boolean isOrganization) {
+    String searchQuery = isOrganization ? String.format(enrichmentQueryOrgs, entityId)
```

Comment:
there was a recent change request to use specific fields for each entity type for counting enrichments
We need to implement this change as well
Agent: edm_agent
Concept: skos_concept
Place: edm_place
Timespan:  edm_timespan
Organization: foaf_organization

See also get_enrichment_count_field() in entity-api-import
https://github.com/europeana/entity-api-import/blob/5e516bc3ac723d4a81e3cb3af991f933584aa247/entity_collection/munge/mongo_import/entities/ranking_metrics/RelevanceCounter.py#L159